src/backend/d3d12/CommandBufferD3D12.cpp

```diff
@@ -258,20 +259,74 @@ namespace d3d12 {
                         BeginRenderPassCmd* beginRenderPassCmd = commands.NextCommand<BeginRenderPassCmd>();
                         currentRenderPass = ToBackend(beginRenderPassCmd->renderPass.Get());
                         currentFramebuffer = ToBackend(beginRenderPassCmd->framebuffer.Get());
+                        currentSubpass = 0;
 
                         uint32_t width = currentFramebuffer->GetWidth();
                         uint32_t height = currentFramebuffer->GetHeight();
                         D3D12_VIEWPORT viewport = { 0.f, 0.f, static_cast<float>(width), static_cast<float>(height), 0.f, 1.f };
                         D3D12_RECT scissorRect = { 0, 0, static_cast<long>(width), static_cast<long>(height) };
                         commandList->RSSetViewports(1, &viewport);
                         commandList->RSSetScissorRects(1, &scissorRect);
-                        commandList->OMSetRenderTargets(1, &device->GetCurrentRenderTargetDescriptor(), FALSE, nullptr);
                     }
                     break;
 
                 case Command::BeginRenderSubpass:
                     {
                         commands.NextCommand<BeginRenderSubpassCmd>();
+                        const auto& subpassInfo = currentRenderPass->GetSubpassInfo(currentSubpass);
```

Comment:
Just saw that you did part of that in the next commit, though more things could be moved in d3d12::Framebuffer